### DISCUSSION TITLE
RGBA Alpha of Pixels

### DISCUSSION THREAD

**Discussion by undersquire** [parasyte/pixels#203] [Q&A]
I just had a quick question about something regarding pixels in the pixel frame buffer. I see that we need to write pixels in the frame buffer in groups of 4 bytes, assuming that each byte is one of the RGBA values. However, I've noticed that the last byte, what should be the Alpha value, doesn't seem to do anything to the resulting pixels. Is this on purpose? If so, will it ever support the alpha value?

**Comment by parasyte** [comment]
Hi, I converted this issue to a discussion, which you will be able to find in the discussions tab.

The default texture format is RGBA (sRGB with 8-bits per component), but it can be reconfigured to use any texture format supported by `wgpu`. So, if you like, you can switch to a 24-bit texture format to remove the alpha component.

The reason for defaulting to RGBA (with an alpha component), apart from familiarity, is that it can potentially be used with pre-multiplied alpha colors for blending purposes. This can either be used with a custom post-processing shader, or with transparent windows. Window transparency is currently not fully supported by `wgpu`, but they have an open issue for it: https://github.com/gfx-rs/wgpu/issues/687

**Reply by JMS55** [reply]
I use the alpha component in a post-processing shader to detect which pixels should have the effect applied to it. Quite handy.

**Comment by parasyte** [comment]
Initial alpha blending support was added in #236.
